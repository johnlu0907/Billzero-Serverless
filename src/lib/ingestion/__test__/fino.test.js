/**
 * Tests the finoCredentialsConverter.test.js
 *
 */
const finoConverter = require("../finoConverter");
const utils = require("./utils");

const providerIds = [
  "f8e3653f-da3a-4396-af22-6fbf5bbe1f46",
  "cae2f322-59a6-4592-a569-6559735af17f",
  "cb3055a8-e5d1-4bef-9453-870a40c85117",
  "cf88552b-e5c9-4f97-bc34-f623f31155c7",
  "d2e72d3c-7539-4471-a5ca-4823216b615a",
  "d5789264-ae63-41b2-91da-33be3bbc8a99",
  "c75450a8-f3b5-4066-a2bc-4a84e4b0cdd8",
];

describe("fino converter", () => {
  describe("providerToVendor", () => {
    const providers = utils.getProviders(providerIds);

    providers.map((provider) => {
      describe(`providerId -> ${provider.immutableId}`, () => {
        test("Snapshot matched", () => {
          const vendor = finoConverter.providerToVendor(provider);

          delete vendor.updatedAt;
          delete vendor.createdAt;

          expect(vendor).toMatchSnapshot();
        });

        test("Attribute type checking", () => {
          const vendor = finoConverter.providerToVendor(provider);

          expect(vendor.supported).toBe("false");
          expect(typeof vendor.updatedAt).toBe("string");
          expect(typeof vendor.createdAt).toBe("string");
        });
      });
    });
  });

  test("isPayable", () => {
    const payableProvider = utils.getProviders([
      "cb3055a8-e5d1-4bef-9453-870a40c85117",
    ])[0];
    const notPayableProvider = utils.getProviders([
      "cae2f322-59a6-4592-a569-6559735af17f",
    ])[0];

    expect(finoConverter.isPayable(payableProvider)).toBe(true);
    expect(finoConverter.isPayable(notPayableProvider)).toBe(false);
  });
});
