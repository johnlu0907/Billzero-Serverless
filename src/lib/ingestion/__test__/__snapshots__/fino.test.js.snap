// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fino converter providerToVendor providerId -> c75450a8-f3b5-4066-a2bc-4a84e4b0cdd8 Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "utilities",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "Email Address",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "c75450a8-f3b5-4066-a2bc-4a84e4b0cdd8",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "villageofwalworth.utilities.png",
  "name": "Village of Walworth, Wisconsin",
  "payMethods": Array [
    "cc",
  ],
  "prefix": "village of walworth, wisconsin",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "village of walworth, wisconsin",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> cae2f322-59a6-4592-a569-6559735af17f Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "bank",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "Member Number",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "cae2f322-59a6-4592-a569-6559735af17f",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "bank-icon.png",
  "name": "Corporate America Family CU (US)",
  "payMethods": Array [],
  "prefix": "corporate america family cu (us)",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "corporate america family cu (us)",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> cb3055a8-e5d1-4bef-9453-870a40c85117 Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "creditCard",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "User Name",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "cb3055a8-e5d1-4bef-9453-870a40c85117",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "mavisdiscounttire.creditcard.png",
  "name": "Mavis Discount Tire",
  "payMethods": Array [
    "ach",
  ],
  "prefix": "mavis discount tire",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "mavis discount tire",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> cf88552b-e5c9-4f97-bc34-f623f31155c7 Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "bank",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "User ID",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "cf88552b-e5c9-4f97-bc34-f623f31155c7",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "bank-icon.png",
  "name": "Independent Bank, Michigan",
  "payMethods": Array [],
  "prefix": "independent bank, michigan",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "independent bank, michigan",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> d2e72d3c-7539-4471-a5ca-4823216b615a Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "bank",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "Account Number / User Name",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "d2e72d3c-7539-4471-a5ca-4823216b615a",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "bank-icon.png",
  "name": "Grow FinancialFCU Personal Portal (US)",
  "payMethods": Array [],
  "prefix": "grow financialfcu personal portal (us)",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "grow financialfcu personal portal (us)",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> d5789264-ae63-41b2-91da-33be3bbc8a99 Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "utilities",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "Username",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "d5789264-ae63-41b2-91da-33be3bbc8a99",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "passaicvalleywatercommission.utilities.png",
  "name": "Passaic Valley Water Commission",
  "payMethods": Array [],
  "prefix": "passaic valley water commission",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "passaic valley water commission",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;

exports[`fino converter providerToVendor providerId -> f8e3653f-da3a-4396-af22-6fbf5bbe1f46 Snapshot matched 1`] = `
Object {
  "bgColor": "ffffff",
  "bill_type": "credentials",
  "biller_type": "loan",
  "bztype": Array [],
  "can_migrate": false,
  "change_user": false,
  "country": "US",
  "credentials": Array [
    Object {
      "codeList": false,
      "masked": false,
      "name": "User ID",
      "orderOfDisplay": 0,
    },
    Object {
      "codeList": false,
      "masked": true,
      "name": "Password",
      "orderOfDisplay": 1,
    },
  ],
  "currency": "USD",
  "has_xdata": "true",
  "id": "f8e3653f-da3a-4396-af22-6fbf5bbe1f46",
  "image": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "imagex": "https://billzero-prod.s3.amazonaws.com/vendors/imageDefault.jpg",
  "login_category": "true",
  "logo": "loan-icon.png",
  "name": "Tompkins State Bank (US)",
  "payMethods": Array [],
  "prefix": "tompkins state bank (us)",
  "required_parameters": Array [],
  "returned_parameters": Array [],
  "sname": "tompkins state bank (us)",
  "supported": "false",
  "textColor": "ffffff",
  "topVendorIndex": 100,
  "type": "biller",
  "xPayBillerIds": Array [],
}
`;
